pkg/storaged/plot.jsx

```diff
@@ -0,0 +1,271 @@
+                        <li role="presentation">
+                            <a role="menuitem" tabindex="-1" data-range="300">{_("5 minutes")}</a>
+                        </li>
+                        <li role="presentation">
+                            <a role="menuitem" tabindex="-1" data-range="3600">{_("1 hour")}</a>
+                        </li>
+                        <li role="presentation">
+                            <a role="menuitem" tabindex="-1" data-range="21600">{_("6 hours")}</a>
+                        </li>
+                        <li role="presentation">
+                            <a role="menuitem" tabindex="-1" data-range="86400">{_("1 day")}</a>
+                        </li>
+                        <li role="presentation">
+                            <a role="menuitem" tabindex="-1" data-range="604800">{_("1 week")}</a>
+                        </li>
+                    </ul>
+                </div>
+                { "\n" }
+                <button className="btn btn-default" data-action="zoom-out">
+                    <span className="glyphicon glyphicon-zoom-out"></span>
+                </button>
+                { "\n" }
+                <div className="btn-group">
+                    <button className="btn btn-default fa fa-angle-left" data-action="scroll-left"></button>
+                    <button className="btn btn-default fa fa-angle-right" data-action="scroll-right"></button>
+                </div>
+            </div>
+        );
+    }
+
+    reset(plots) {
+        if (this.controls)
+            this.controls.reset(plots);
+        else
+            this.plots = plots;
+    }
+}
+
+class StoragePlot extends React.Component {
+    constructor() {
+        super();
+        this.state = { unit: "" }
+        this.on_resize = () => { this.setState({}); };
+    }
+
+    componentDidMount() {
+        window.addEventListener("resize", this.on_resize);
+    }
+
+    componentWillUnmount() {
+        window.removeEventListener("resize", this.on_resize);
+    }
+
+    render() {
+        var self = this;
+
+        function setup_hook(flot) {
+            var axes = flot.getAxes();
+            if (axes.yaxis.datamax < 100000)
+                axes.yaxis.options.max = 100000;
+            else
+                axes.yaxis.options.max = null;
+            axes.yaxis.options.min = 0;
+        }
+
+        function post_hook(flot) {
+            var axes = flot.getAxes();
+            self.setState({ unit: cockpit.format_bytes_per_sec(axes.yaxis.max, 1024, true)[1] });
+        }
+
+        function hover(event, dev) {
+            if (self.props.onHover)
+                self.props.onHover(dev);
+        }
+
+        function setup_plot(element) {
+            if (!element)
+                return;
+
+            if (element != self.last_element) {
+                if (self.last_element) {
+                    console.log("REINIT!");
+                    // XXX - reset plot and create new
```

Comment:
Does this still need to be implemented in this PR or is this not a regression?

Reply:
Ah, yes.  This situation _should_ not happen, but the guarantee is fragile.  So I should implement this.

Reply:
Done.